tests/tests/budget_service_tests.cpp

```diff
@@ -237,45 +494,144 @@ BOOST_AUTO_TEST_CASE(allocate_cash_next_block)
         db_plugin->debug_update(
             [=](database&) {
                 asset balance(BUDGET_BALANCE_DEFAULT, SCORUM_SYMBOL);
-                time_point_sec deadline(time_point_sec::maximum());
+                time_point_sec deadline(default_deadline);
 
-                budget_service.create_fund_budget(balance, BUDGET_PER_BLOCK_DEFAULT, deadline);
+                budget_service.create_fund_budget(balance, deadline);
 
             },
             default_skip);
 
         generate_block();
 
         {
-            const auto& budget = budget_service.get_any_budget(SCORUM_ROOT_POST_PARENT);
+            auto budgets = budget_service.get_fund_budgets();
+
+            BOOST_REQUIRE(!budgets.empty());
+
+            const budget_object& budget = budgets[0];
 
             auto cash = budget_service.allocate_cash(budget);
 
             BOOST_REQUIRE(cash.amount == BUDGET_PER_BLOCK_DEFAULT);
         }
 
         {
-            const auto& budget = budget_service.get_any_budget(SCORUM_ROOT_POST_PARENT);
+            auto budgets = budget_service.get_fund_budgets();
+
+            BOOST_REQUIRE(!budgets.empty());
+
+            const budget_object& budget = budgets[0];
 
             BOOST_REQUIRE(budget.balance.amount == (BUDGET_BALANCE_DEFAULT - BUDGET_PER_BLOCK_DEFAULT));
         }
     }
     FC_LOG_AND_RETHROW()
 }
 
-BOOST_AUTO_TEST_CASE(is_const_ref_to_same_memory)
+BOOST_AUTO_TEST_CASE(auto_close_fund_budget_by_deadline)
+{
+    try
+    {
+        db_plugin->debug_update(
```

Comment:
move this code in to the function with a meaningful name. This change will increase readability.